torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1565,18 +1565,31 @@ def _unshard(
         """
         if not handles:
             return
+        training_state = self._get_training_state(tuple(handles))
+        # Do not use the pre-all-gather and all-gather streams if in
+        # `summon_full_params()` to avoid overallocating blocks to that stream
+        pre_unshard_stream = (
```

Comment:
Will defer to your thoughts but determining the stream based on the training state inside _unshard seems like it could be a little confusing / unclear. Have you considered determining the streams to use at a higher layer and passing in the streams to this function? Would that help make this clearer?